### PR TITLE
vWAN: Moved resources to corresponding modules and removed duplicates

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Moved resources to corresponding modules and removed duplicates (incl. RBAC)
- Added new vWAN parameters
- Added new vWAN parameter file

Pipeline reference
[Network: VirtualWans](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.virtualwans.yml)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

### CODE REVIEW

File: arm/Microsoft.Network/virtualWans/deploy.bicep [hunk 1]
```diff
@@ -9,41 +9,16 @@ param name string
...
+param type string = 'Standard'
 
-@description('Optional. Name of the Virtual Hub. A virtual hub is created inside a virtual wan.')
-param virtualHubName string = 'SampleVirtualHub'
+@description('Optional. True if branch to branch traffic is allowed.')
+param allowBranchToBranchTraffic bool = false
 
-@description('Optional. Name of the Vpn Gateway. A vpn gateway is created inside a virtual hub.')
-param vpnGatewayName string = 'SampleVpnGateway'
+@description('Optional. True if branch to branch traffic is allowed.')
```
Comment (eriqua) [inline arm/Microsoft.Network/virtualWans/deploy.bicep:17]:
```suggestion
@description('Optional. True if Vnet to Vnet traffic is allowed. ')
```

File: arm/Microsoft.Network/virtualWans/deploy.bicep [hunk 2]
```diff
@@ -9,41 +9,16 @@ param name string
...
-@description('Optional. The hub address prefix. This address prefix will be used as the address prefix for the hub vnet')
-param addressPrefix string = '192.168.0.0/24'
-
-@description('Optional. his needs to be set to true if BGP needs to enabled on the vpn connection.')
-@allowed([
-  'true'
-  'false'
-])
-param enableBgp string = 'false'
+@description('Optional. True if branch to branch traffic is allowed.')
```
Comment (eriqua) [inline arm/Microsoft.Network/virtualWans/deploy.bicep:20]:
```suggestion
@description('Optional. Vpn encryption to be disabled or not.')
```

File: arm/Microsoft.Network/virtualWans/readme.md [hunk 3]
```diff
@@ -8,32 +8,22 @@ This template deploys a virtual WAN.
...
+| `Microsoft.Network/virtualWans` | 2021-05-01 |
 
 ## Parameters
 
 | Parameter Name | Type | Default Value | Possible Values | Description |
 | :-- | :-- | :-- | :-- | :-- |
-| `addressPrefix` | string | `192.168.0.0/24` |  | Optional. The hub address prefix. This address prefix will be used as the address prefix for the hub vnet |
-| `connectionName` | string | `SampleVpnsiteVpnGwConnection` |  | Optional. Name of the vpnconnection. A vpn connection is established between a vpnsite and a vpn gateway. |
+| `allowBranchToBranchTraffic` | bool | `False` |  | Optional. True if branch to branch traffic is allowed. |
+| `allowVnetToVnetTraffic` | bool | `False` |  | Optional. True if branch to branch traffic is allowed. |
```
Comment (eriqua) [inline arm/Microsoft.Network/virtualWans/readme.md:18]: Please update after changes

File: arm/Microsoft.Network/virtualWans/deploy.bicep [hunk 4]
```diff
@@ -9,41 +9,16 @@ param name string
   'Standard'
   'Basic'
 ])
-param virtualWanSku string = 'Standard'
+param type string = 'Standard'
```
Comment (eriqua) [inline arm/Microsoft.Network/virtualWans/deploy.bicep:12]: I'd suggest changing the type description to "Optional. The type of the VirtualWAN."

File: arm/Microsoft.Network/virtualWans/readme.md (hunk 3 above)
Comment (MrMCake) [reply arm/Microsoft.Network/virtualWans/readme.md]: done

File: arm/Microsoft.Network/virtualWans/deploy.bicep (hunk 4 above)
Comment (MrMCake) [reply arm/Microsoft.Network/virtualWans/deploy.bicep:12]: done

File: arm/Microsoft.Network/virtualWans/deploy.bicep [hunk 5]
```diff
@@ -1,10 +1,10 @@
 @description('Optional. Location where all resources will be created.')
 param location string = resourceGroup().location
 
-@description('Required. Name of the Virtual Wan.')
+@description('Required. Name of the Virtual VNET.')
```
Comment (eriqua) [inline arm/Microsoft.Network/virtualWans/deploy.bicep:4]: I see VNET, is that correct?

Comment (MrMCake) [reply arm/Microsoft.Network/virtualWans/deploy.bicep]: Fixed

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**